packages/kit/src/exports/public.d.ts

```diff
@@ -45,6 +45,11 @@ export interface Adapter {
 		 */
 		read?: (details: { config: any; route: { id: string } }) => boolean;
 	};
+	/**
+	 * This function is used to allow the adapter to emulate the platform object during dev and
+	 * preview
+	 */
+	emulatePlatform?(svelteKitConfig: KitConfig): MaybePromise<App.Platform>;
```

Comment:
I don't _think_ we need to pass the config into here, unless you had something specific in mind? If we do need it, it should be `opts: { config: KitConfig }` so that we can add other stuff in future if necessary

```suggestion
	emulatePlatform?(): MaybePromise<App.Platform>;
```

Reply:
It was just for future proofing the solution/allowing other adapters to add things on top of it (which your solution addresses better)

Anyways it is not needed and it can be omitted entirely if you prefer (and/or omitted for now and potentially added when/if needed)